demisto_sdk/commands/format/tests/test_formatting_json_test.py

```diff
@@ -622,6 +627,62 @@ def test_set_output_path(self, invalid_path_layouts_formatter):
         assert expected_path == invalid_path_layouts_formatter.output_file
 
 
+class TestFormattingPreProcessRule:
+
+    @pytest.fixture(autouse=True)
+    def pre_process_rules_copy(self):
+        os.makedirs(PRE_PROCESS_RULES_PATH, exist_ok=True)
+        yield shutil.copyfile(SOURCE_FORMAT_PRE_PROCESS_RULES_COPY, DESTINATION_FORMAT_PRE_PROCESS_RULES_COPY)
+        os.remove(DESTINATION_FORMAT_PRE_PROCESS_RULES_COPY)
+        os.rmdir(PRE_PROCESS_RULES_PATH)
+
+    @pytest.fixture(autouse=True)
+    def pre_process_rules_formatter(self, pre_process_rules_copy):
+        yield PreProcessRulesBaseFormat(input=pre_process_rules_copy, output=DESTINATION_FORMAT_PRE_PROCESS_RULES_COPY)
+
+    @pytest.fixture(autouse=True)
+    def invalid_path_pre_process_rules_formatter(self, pre_process_rules_copy):
+        yield PreProcessRulesBaseFormat(input=pre_process_rules_copy, output=DESTINATION_FORMAT_PRE_PROCESS_RULES_INVALID_NAME_COPY)
+
+    def test_remove_unnecessary_keys(self, pre_process_rules_formatter):
+        """
+        Given
+            - A pre_process_rule file with fields that dont exit in pre_process_rule schema.
+        When
+            - Run format on pre_process_rule file
+        Then
+            - Ensure that unnecessary keys were removed
+        """
+        pre_process_rules_formatter.schema_path = PRE_PROCESS_RULES_SCHEMA_PATH
+        pre_process_rules_formatter.remove_unnecessary_keys()
+        for field in ['quickView', 'sortValues', 'someFieldName']:
+            assert field not in pre_process_rules_formatter.data
+
+    def test_set_description(self, pre_process_rules_formatter):
+        """
+        Given
+            - A pre_process_rule file without a description field
+        When
+            - Run format on pre_process_rule file
+        Then
+            - Ensure that description field was updated successfully with '' value
+        """
+        pre_process_rules_formatter.set_description()
+        assert 'description' in pre_process_rules_formatter.data
+
+    def test_set_toVersion(self, pre_process_rules_formatter):
+        """
+        Given
+            - A pre_process_rule file without a toVersion field
+        When
+            - Run format on pre_process_rule file
+        Then
+            - Ensure that toVersion field was updated successfully with '5.9.9' value
+        """
+        pre_process_rules_formatter.set_toVersion()
+        assert pre_process_rules_formatter.data.get('toVersion') == '5.9.9'
```

Comment:
this is not valid.
the pre-processing rule cannot have from-version/to-version lower than 6.5.

Reply:
This field is irrelevant for PreProcessRule.
https://github.com/demisto/demisto-sdk/commit/a0fd82ed05d10a1b093d6d5fc3c8da3bcb433444